### PR TITLE
[URP] Remove unnecessary CopyDepth pass before post processing

### PR DESCRIPTION
**Pull request by manuele-bonanno**
…y for the post processing pass

# **Please read the [Contributing guide](CONTRIBUTING.md) before making a PR.**

* Read the [Graphics repository & Yamato FAQ](http://go/graphics-yamato-faq).

### Checklist for PR maker
- [ ] Have you added a backport label (if needed)? For example, the `need-backport-*` label. After you backport the PR, the label changes to `backported-*`.
- [ ] Have you updated the changelog? Each package has a `CHANGELOG.md` file.
- [ ] Have you updated or added the documentation for your PR? When you add a new feature, change a property name, or change the behavior of a feature, it's best practice to include related documentation changes in the same PR. If you do add documentation, make sure to add the relevant Graphics Docs team member as a reviewer of the PR. If you are not sure which person to add, see the [Docs team contacts sheet](https://docs.google.com/spreadsheets/d/1rgUWWgwLFEHIQ3Rz-LnK6PAKmbM49DZZ9al4hvnztOo/edit#gid=1058860420).
- [ ] Have you added a graphic test for your PR (if needed)? When you add a new feature, or discover a bug that tests don't cover, please add a graphic test.

---
### Purpose of this PR
Why is this PR needed, what hard problem is it solving/fixing?

---
### Testing status
Describe what manual/automated tests were performed for this PR

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

File: com.unity.render-pipelines.universal/Runtime/UniversalRenderer.cs [hunk 1]
```diff
@@ -519,6 +519,12 @@ public override void Setup(ScriptableRenderContext context, ref RenderingData re
                 createColorTexture = createDepthTexture;
             }
 
+            bool requiresDepthCopyPass = !requiresDepthPrepass
+                && requiresDepthTexture
+                && createDepthTexture;
+            // Post process passes that require depth texture (Motion Blur / DoF) as shader read resource will need to access depth information from main geometry pass if there is no depth copy
+            // and depth pre-pass enqueued
+            bool postProcessingUseDepthTextureMainPass = !requiresDepthCopyPass && !requiresDepthPrepass && cameraHasPostProcessingWithDepth;
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/Runtime/UniversalRenderer.cs:527]:
```suggestion
            bool postProcessingUseDepthTextureMainPass = !requiresDepthCopyPass && cameraHasPostProcessingWithDepth;
```
 
 `requiresDepthPrepass` is reduntant logic here. If you expand `requiresDepthCopyPass` the expression you get:

bool postProcessingUseDepthTExtureMainPass = (!**requiresDepthPrepass** && requiresDepthTexture && createDepthTexture) && !**requiresDepthPrepass** && cameraHasPostProcessingWithDepth;

File: com.unity.render-pipelines.universal/Runtime/UniversalRenderer.cs [hunk 2]
```diff
@@ -684,13 +687,13 @@ public override void Setup(ScriptableRenderContext context, ref RenderingData re
...
                 // handle multisample depth resolve by setting the appropriate store actions if supported
-                if (requiresDepthCopyPass && cameraTargetDescriptor.msaaSamples > 1 && RenderingUtils.MultisampleDepthResolveSupported(useRenderPassEnabled))
+                if ((requiresDepthCopyPass || postProcessingUseDepthTextureMainPass) && cameraTargetDescriptor.msaaSamples > 1 && RenderingUtils.MultisampleDepthResolveSupported(useRenderPassEnabled))
                 {
                     bool isCopyDepthAfterTransparent = m_CopyDepthPass.renderPassEvent == RenderPassEvent.AfterRenderingTransparents;
 
                     // we could StoreAndResolve when the depth copy is after opaque, but performance wise doing StoreAndResolve of depth targets is more expensive than a simple Store + following depth copy pass on Apple GPUs,
                     // because of the extra resolve step. So, unless we are copying the depth after the transparent pass, just Store the depth target.
-                    if (isCopyDepthAfterTransparent && !copyColorPass)
+                    if ((isCopyDepthAfterTransparent || postProcessingUseDepthTextureMainPass) && !copyColorPass)
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/Runtime/UniversalRenderer.cs:696]: Nitpicking: you have in multiple places isCopyDEpthAfterTransparent || postProcessingUseDepthTextureMainPass maybe you could give have a variable with some meaning for this to improve readability. The logic of store actions is a little bit convoluted, but I guess the plan is to improve it with RenderGraph anyway, so no big deal.

**Review by phi-lira — APPROVED** [review]
Missing changelogs, otherwise minor comments that should not block PR.

**Comment by manuele-bonanno** [comment]
this is WIP and not currentlly being worked on. Closing the PR temporarily to avoid confusion

**Pull request closed without merging**